spec/rubocop/cop/lint/useless_trailing_comma_spec.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::UselessTrailingComma do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  context 'curly brace block format' do
+    it 'registers an offense when a trailing comma is not needed' do
+      expect_offense(<<~RUBY)
+        test { |a, b,| a + b }
+               ^^^^^^^ Useless trailing comma present in block arguments.
+      RUBY
+    end
+
+    it 'does not register an offense when a trailing comma is required' do
+      expect_no_offenses(<<~RUBY)
+        test { |a,| a }
+      RUBY
+    end
+
+    it 'does not register an offense when no arguments are present' do
+      expect_no_offenses(<<~RUBY)
+        test { a }
+      RUBY
+    end
+
+    it 'does not register an offense when more than one argument is ' \
+       'present with no trailing comma' do
+      expect_no_offenses(<<~RUBY)
+        test { |a, b| a + b }
+      RUBY
+    end
+
+    it 'does not register an offense for default arguments' do
+      expect_no_offenses(<<~RUBY)
+        test { |a, b, c = nil| a + b + c }
+      RUBY
+    end
+
+    it 'does not register an offense for keyword arguments' do
+      expect_no_offenses(<<~RUBY)
+        test { |a, b, c: 1| a + b + c }
+      RUBY
+    end
+
+    it 'does not register an offense for a hash literal' do
+      expect_no_offenses(<<~RUBY)
+        {}
+      RUBY
+    end
+
+    it 'auto-corrects useless commas' do
+      new_source = autocorrect_source(<<~RUBY)
+        test { |a, b,| a + b }
+      RUBY
+      expect(new_source).to eq(<<~RUBY)
+        test { |a, b| a + b }
+      RUBY
+    end
+
+    it 'does not auto-correct commas when only one argument is present' do
+      new_source = autocorrect_source(<<~RUBY)
+        test { |a,| a }
+      RUBY
+      expect(new_source).to eq(<<~RUBY)
+        test { |a,| a }
+      RUBY
+    end
+
+    it 'does not auto-correct curly brace blocks without arguments' do
+      new_source = autocorrect_source(<<~RUBY)
+        test { a }
+      RUBY
+      expect(new_source).to eq(<<~RUBY)
+        test { a }
+      RUBY
+    end
+
+    it 'does not auto-correct hash literals' do
```

Comment:
I'm not sure that you need this test since hash literals won't register an offense.